luaparse.js

```diff
@@ -907,15 +910,27 @@
           // Return it as is, without translating the byte.
           return '\\' + input.slice(sequenceStart, index);
         }
+        return raise({}, errors.hexadecimalDigitExpected, '\\' + input.slice(sequenceStart, index + 2));
+      case '\\':
+      case '\'':
+      case '"':
+      case '[':
+      case ']':
         return '\\' + input.charAt(index++);
       default:
         // \ddd, where ddd is a sequence of up to three decimal digits.
-        if (isDecDigit(input.charCodeAt(index))) {
-          while (isDecDigit(input.charCodeAt(++index)));
-          return '\\' + input.slice(sequenceStart, index);
+        while (isDecDigit(input.charCodeAt(index)) && index - sequenceStart < 3) index++;
+
+        if (index === sequenceStart) {
+          raise({}, errors.invalidEscape, '\\' + input.slice(sequenceStart, index + 1));
```

Comment:
Makes sense, I added a luaVersion option, so by default 5.1 this error wont raise.